### DISCUSSION TITLE
[JSONAPI] Nested resource inclusion not exposing specified properties

### DISCUSSION THREAD

**Discussion by maximgubar** [api-platform/core#5143] [Q&A]
I have 2 entities, Product and Supplier

```php
<?php
/**
 * @ApiResource(
 *     itemOperations={
 *         "get"={
 *             "normalization_context"={"groups"={"single-read"}},
 *         },
 *     },
 * )
 * @ORM\Table(name="product")
 * @ORM\Entity(repositoryClass="App\Repository\ProductRepository")
 * @ApiFilter(SearchFilter::class, properties={"id": "exact", "supplier": "exact"})
 */
class Product
{
    /**
     * @ApiProperty(identifier=true)
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\ManyToOne(targetEntity="Supplier", inversedBy="products")
     * @ORM\JoinColumn(name="supplier_id", referencedColumnName="id")
     * @Groups({"single-read"})
     */
    private $supplier;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups({"single-read"})
     */
    private $sku;
}
```

```php
<?php

/**
 * @ORM\Table(name="supplier")
 * @ORM\Entity(repositoryClass="App\Repository\SupplierRepository")
 * @ApiResource(
 *     attributes={
 *         "normalization_context"={"groups"={"read"}},
 *         "denormalization_context"={"groups"={"write"}},
 *     },
 * )
 * @ApiFilter(SearchFilter::class, properties={"id": "exact", "code": "exact", "name": "exact"})
 */
class Supplier
{
    /**
     * @ApiProperty(identifier=true)
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255, unique=true)
     * @Groups({"read", "write", "single-read"})
     */
    private $code;

    /**
     * @ORM\Column(type="string", length=255, unique=true)
     * @Groups({"read", "write", "single-read"})
     */
    private $name;
}
```

When I use 'application/json' as response content type, properties 'code' and 'name' are exposed.
If I'm using 'application/vnd.api+json' I always see dereferencable IRI, 
```json
"relationships": {
      "supplier": {
        "data": {
          "type": "Supplier",
          "id": "/api/suppliers/73"
        }
      }
    }
```

Am I missing something in the configuration or this is a bug ?

**Comment by teohhanhui** [comment]
Please see https://jsonapi.org/format/#document-resource-object-linkage

It's correct according to spec. You have to find them under the `included` top-level key: https://jsonapi.org/format/#document-top-level

> An endpoint **MAY** return resources related to the primary data by default.

If I understand correctly, it doesn't seem that API Platform has chosen to do that (see https://github.com/api-platform/core/blob/v2.4.2/features/jsonapi/related-resouces-inclusion.feature), so you have to use the `include` query parameter: https://jsonapi.org/format/#fetching-includes

But perhaps we should include related resources according to the serializer groups by default. It seems more correct to me.

**Comment by maximgubar** [comment]
@teohhanhui thanks for the clarification :)

**Comment by teohhanhui** [comment]
Re-opening for this:

> But perhaps we should include related resources according to the serializer groups by default. It seems more correct to me.

**Comment by teohhanhui** [comment]
> But perhaps we should include related resources according to the serializer groups by default. It seems more correct to me.

~That's already the case. Sorry.~ I spoke too soon.
